R/ard_dichotomous.R

```diff
@@ -126,22 +126,17 @@ maximum_variable_values <- function(data) {
   imap(
     values,
     function(value, column) {
-      if (length(value) != 1L || is_empty(value) || is.na(value) || is.nan(value) || is.infinite(value)) {
-        cli::cli_abort(c(
-          "Error in argument {.arg values} for variable {.val {column}}.",
-          "i" = "The length of the value must be one and not one of {.val {c(NA, NaN, Inf)}}."
-        ), call = call)
-      }
-      if (inherits(data[[column]], "factor") && !value %in% levels(data[[column]])) {
-        cli::cli_abort(c(
-          "Error in argument {.arg values} for variable {.val {column}}.",
-          "i" = "A value of {.val {value}} was passed, but must be one of {.val {levels(data[[column]])}}."
-        ), call = call)
-      } else if (!value %in% data[[column]]) {
-        cli::cli_abort(c(
-          "Error in argument {.arg values} for variable {.val {column}}.",
-          "i" = "A value of {.val {value}} was passed, but must be one of {.val {unique(data[[column]]) |> na.omit() |> sort()}}."
-        ), call = call)
+      accepted_values <- .unique_and_sorted(data[[column]])
```

Comment:
should this also include unobserved factor levels?

Reply:
I think this should include unobserved levels, because otherwise there would be no way to have dichotomous summaries unless there have been those values observed?

Reply:
Agree, sorry I thought that .unique_and_sorted only included observed factor levels, you can ignore :)